pkg/ui/src/views/shared/components/toolTip/index.tsx

```diff
@@ -22,15 +24,38 @@ interface ToolTipWrapperState {
  * contents.
  */
 export class ToolTipWrapper extends React.Component<ToolTipWrapperProps, ToolTipWrapperState> {
+  popperInstance: Popper;
+  content: ReactInstance;
+  text: ReactInstance;
+
   constructor(props?: ToolTipWrapperProps, context?: any) {
     super(props, context);
     this.state = {
       hovered: false,
     };
   }
 
+  componentWillUnmount() {
+    if (this.popperInstance) {
+      this.popperInstance.destroy();
+    }
+  }
+
+  initPopper() {
+    const contentEl =  ReactDOM.findDOMNode(this.content) as Element;
+    const tooltipEl =  ReactDOM.findDOMNode(this.text) as Element;
+
+    // PopperOptions.eventsEnabled should be set to `false` to prevent
+    // performance issues on pages with a large number of tooltips
+    this.popperInstance = new Popper(contentEl, tooltipEl, {
+      placement: "auto",
+      eventsEnabled: false
+    });
+  }
+
   onMouseEnter = () => {
     this.setState({hovered: true});
```

Comment:
I think we can get rid of the `setState` calls now to skip the React render cycle.

Reply:
When I remove the `setState` calls there isn't anything to manage the show/hide behavior of the tooltip.  Am I misinterpreting what you're suggesting?

Reply:
Doesn't popper.js handle that now?  Maybe I'm confused about exactly what functionality it offers.

Reply:
Yeah PopperJS is just the positioning engine whereas TooltipJS is a frontend that implements Popper.